cmd/dependencycheck/dependencycheck.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Checks for restricted dependencies in go packages.
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"regexp"
+)
+
+var (
+	exclude  = flag.String("exclude", "", "skip packages regex pattern (e.g. '^k8s.io/kubernetes/')")
+	restrict = flag.String("restrict", "", "restricted dependencies regex pattern (e.g. '^k8s.io/(apimachinery|client-go)/')")
+)
+
+type goPackage struct {
+	Name       string
+	ImportPath string
+	Deps       []string
```

Comment:
Deps includes all transitive dependencies of this package, which is likely to be enormous

Imports + TestImports + XTestImports is likely to be more useful in identifying the actual problematic import that needs to be dealt with

suggest:
```diff
--- a/cmd/dependencycheck/dependencycheck.go
+++ b/cmd/dependencycheck/dependencycheck.go
@@ -34,9 +34,11 @@ var (
 )
 
 type goPackage struct {
-       Name       string
-       ImportPath string
-       Deps       []string
+       Name         string
+       ImportPath   string
+       Imports      []string
+       TestImports  []string
+       XTestImports []string
 }
 
 func main() {
@@ -86,7 +88,12 @@ func main() {
                        continue
                }
                importViolations := []string{}
-               for _, d := range p.Deps {
+
+               allImports := []string{}
+               allImports = append(allImports, p.Imports...)
+               allImports = append(allImports, p.TestImports...)
+               allImports = append(allImports, p.XTestImports...)
+               for _, d := range allImports {
                        if depsPattern.MatchString(d) {
                                importViolations = append(importViolations, d)
                        }
```